flexmeasures/data/tests/test_dict_price_sensors.py

```diff
@@ -0,0 +1,84 @@
+from datetime import datetime, timedelta
+import pytest
+import pytz
+from flexmeasures.data.models.planning.storage import (
+    StorageScheduler,
+)
+from flexmeasures.utils.calculations import integrate_time_series
+
+
+TOLERANCE = 0.00001
+
+
+@pytest.mark.parametrize(
+    "roundtrip_efficiency",
+    [
+        0.90,
+        0.95,
+    ],
+)
+def test_schedule(
+    create_solar_plants, create_building, flexible_devices, roundtrip_efficiency: float
+):
+    """
+    Check battery scheduling results for tomorrow
+    """
+    battery = flexible_devices["battery-power"]
+    tz = pytz.timezone("Europe/Amsterdam")
+    start = tz.localize(datetime(2015, 1, 1))
+    end = tz.localize(datetime(2015, 1, 2))
+    resolution = timedelta(minutes=60)
+    soc_at_start = 545
+    soc_min = 0.5
+    soc_max = 795
+    inflexible_devices = [
+        create_solar_plants["solar-power-1"].id,
+        create_building["building-power"].id,
+    ]
+    consumption_price_sensor_per_device = {
+        flexible_devices["Grid-power"]
+        .id: flexible_devices["grid-consumption-price-sensor"]
+        .id
+    }
+    production_price_sensor_per_device = {
+        create_solar_plants["solar-power-1"]
+        .id: create_solar_plants["solar1-production-price-sensor"]
+        .id,
+        flexible_devices["Grid-power"]
+        .id: flexible_devices["grid-production-price-sensor"]
+        .id,
+    }
+    scheduler = StorageScheduler(
+        battery,
+        start,
+        end,
+        resolution,
+        flex_model={
+            "soc-at-start": soc_at_start,
+            "soc-unit": "MWh",
+            "soc-min": soc_min,
+            "soc-max": soc_max,
+            "roundtrip-efficiency": roundtrip_efficiency,
+        },
+        flex_context={
+            "inflexible-device-sensors": inflexible_devices,
+            "consumption-price-sensor-per-device": consumption_price_sensor_per_device,
+            "production-price-sensor-per-device": production_price_sensor_per_device,
+        },
+    )
+    schedule = scheduler.compute()
+    soc_schedule = integrate_time_series(
+        schedule,
+        soc_at_start,
+        up_efficiency=roundtrip_efficiency**0.5,
+        down_efficiency=roundtrip_efficiency**0.5,
+        decimal_precision=5,
+    )
+    # Check if constraints were met
```

Comment:
Still, this is not checking that the impact of having different sensors in the final schedule.  

What about using different price sensor for the storage in two different schedule runs and then compare the two schedules? What do you think @Flix6x ?

Reply:
> Still, this is not checking that the impact of having different sensors in the final schedule.
> 
> What about using different price sensor for the storage in two different schedule runs and then compare the two schedules? What do you think @Flix6x ?

Also what can be done for this?

Reply:
You could come up with two example problems with expected optimal schedules. Maybe including assert statement against the total costs of the two schedules (by multiplying the resulting schedule against corresponding prices), or asserting that a certain effect can be observed in the schedule at a certain time. The two problems could differ in terms of e.g. the size of the PV subsidy, such that with a high subsidy the schedule can be seen to favor feeding PV into the grid, whereas with a low subsidy, the schedule could favor self-consumption instead.

I'm not sure if this particular example makes sense in the context of your use case, but there must be some effect you are hoping to accomplish by setting individual device prices, right? For now this test only checks whether the resulting schedule respects the power constraints and SoC constraints, which does not yet constitute a test of the new functionality.

Reply:
Ok sure